vote.py

```diff
@@ -0,0 +1,172 @@
+            .format(penalty, user)
+
+    # TODO: put all these commands into one regexp match
+    @cmd
+    def votekick(self, event):
+        return self.vote_punishment(event, 'kick')
+
+    @cmd
+    def voteban(self, event):
+        return self.vote_punishment(event, 'ban')
+
+    @cmd
+    def voteunban(self, event):
+        return self.vote_punishment(event, 'unban')
+
+    @cmd
+    def votemute(self, event):
+        return self.vote_punishment(event, 'mute')
+
+    def yesnoquestion_result(self, event=None):
+        """Return results of yes/no question"""
+        yes = []
+        no = []
+        for k, v in self.voted.iteritems():
+            if v[1]:
+                yes.append(v[0])
+            elif not v[1]:
+                no.append(v[0])
+
+        res = '{0}x yes ({2}), {1}x no ({3})'.format(self.yeses,
+                                                     self.nos,
+                                                     ', '.join(yes),
+                                                     ', '.join(no))
+        self.reset()
+        return 'Results of {0}\'s question: {1}'.format(event.meta['nick'],
+                                                        res)
+
+    @cmd
+    def askyesno(self, event):
+        """Ask a yes/no question"""
+        if self.voting or self.voting_q:
+            return 'Voting in progress...'
+
+        args = event.args
+        if not len(args):
+            return 'No question specified.'
+
+        question = ' '.join(args)
+        if not question.endswith('?'):
+            question += '?'
+
+        user = event.meta['nick']
+        self.voting = True
+
+        self.delay_task(60, self.yesnoquestion_result, event=event)
+
+        return '{0} asks: {1} (!yes / !no)'.format(user, question)
+
+    def question_result(self, event=None):
+        """Return results of question"""
+        res = ''
+        for k, v in self.voted.iteritems():
+            res += '{0}->{1}, '.format(v[0], v[1])
+
+        self.reset()
+        return 'Results of {0}\'s question: {1}'.format(event.meta['nick'],
+                                                        res)
+
+    @cmd
+    def ask(self, event):
+        """Ask a question"""
+        if self.voting or self.voting_q:
+            return 'Voting in progress...'
+
+        args = event.args
+        if not len(args):
+            return 'No question specified.'
+
+        question = ' '.join(event.args)
+        if not question.endswith('?'):
+            question += '?'
+
+        user = event.meta['nick']
+        self.voting_q = True
+
+        self.delay_task(60, self.question_result, event=event)
+
+        return '{0} asks: {1} (!vote <answer>)'.format(user, question)
+
+    @cmd
+    def vote(self, event):
+        """Add user's own answer
+        Only usable when answering to !askquestion"""
+        if not self.voting_q:
+            return 'No question to answer in progress...'
```

Comment:
I do understand but that's just because I read the source. Otherwise I'd be just lost when presented with a sentence like this. Could you please rephrase this sentence?